skimage/filters/thresholding.py

```diff
@@ -1233,3 +1233,141 @@ def threshold_multiotsu(image, classes=3, nbins=256):
+    p0 : ndarray
+        weighted mean of all elemnts of x below each split, size = len(x) -1
+    p1 : ndarray
+        weighted mean of all elemnts of x above each split, size = len(x) -1
+    mu0: ndarray
+        weighted distrotion  of all elemnts of x below each split,
+        size = len(x) -1
+    mu1: ndarray
+        weighted distrotion of all elemnts of x above each split,
+        size = len(x) -1
+    d0 : ndarray
+        complex definition
+    d1 : ndarray
+        complex definition
+
+    References
+    -------------------
+    .. [1] A Generalization of Otsu's Method and Minimum Error Thresholding
+            Jonathan T. Barron, ECCV, 2020
+
+    """
+    assert np.all(n >= 0)
+    x = np.arange(len(n), dtype=n.dtype) if x is None else x
+    assert np.all(x[1:] >= x[:-1])
+
+    w0 = _clip(_csum(n))
+    w1 = _clip(_dsum(n))
+    p0 = w0 / (w0 + w1)
+    p1 = w1 / (w0 + w1)
+    mu0 = _csum(n * x) / w0
+    mu1 = _dsum(n * x) / w1
+    d0 = _csum(n * x**2) - w0 * mu0**2
+    d1 = _dsum(n * x**2) - w1 * mu1**2
+    return x, w0, w1, p0, p1, mu0, mu1, d0, d1
+
+
+def theshold_generalized_histogram(n, x=None, nu=0, tau=0, kappa=0, omega=0.5):
+    """Compute the generalized histogram threshold for a histogram
+
+
+    Parameters
+    ----------------------------
+    n : ndarray
+        Each element is the number of pixels falling in each
+         intensity bin.
+    x : ndaarray, optional
+        Each element is the value corresponding to the center of
+         each intensity bin.
+    nu : Float
+        Degree of freedom , positive integer, scaled inverse chi-squared
+         distribution parameter
+    tau : Float
+        scaling parameter, positve integer, scaled inverse
+         chi-squared distribution
+    kappa : Float
+        concentration, positive integer, beta distribution
+    omega : Float
+        mode   , beta distribution ,  0 <= omega <= 1
+
+    Returns
+    ----------------------------
+    t  : float
+        Threshold bin value, corresponding to maximum score
+    score : ndarray
+        all possible values of score for each possible threshold value
+
+
+    References
+    -------------------
+    .. [1] A Generalization of Otsu's Method and Minimum Error Thresholding
+         Jonathan T. Barron, ECCV, 2020
+
+    Examples
+    ----------------
+
+    >>> from skimage.data import camera
+    >>> from skimage.exposure import histogram
+    >>> from skimage.filters import theshold_generalized_histogram
+    >>>
+    >>> data = camera()
+    >>> counts, bin_centers = histogram(data.ravel(), 256, source_range="image")
+    >>>
+    >>> default_nu = np.sum(counts)
+    >>> default_tau = np.sqrt(1/12)
+    >>> default_kappa = np.sum(counts)
+    >>> default_omega = 0.5
+    >>>
+    >>> t, score = theshold_generalized_histogram(counts, bin_centers,
+            default_nu, default_tau, default_kappa, default_omega)
+    >>> binary = data<=t
+    """
+    assert nu >= 0
```

Comment:
Use assertions to self-test the program, not to handle wrong user input. Instead, raise an exception and print an error message. Possibly handle the exception by overwriting wrong input values with the respective default parameter values?